includes/admin/extensions/abstract-extension.php

```diff
@@ -0,0 +1,284 @@
+					'href'        => $this->get_upgrade_url( $product_data, $item_id ),
+					'new_tab'     => true,
+					'type'        => $type,
+				);
+			}
+		} elseif ( ! $this->manager->is_plugin_active( $product_data['basename'] ) ) {
+			// If the extension is installed, but not activated, the button will prompt to activate it.
+			$button['plugin'] = $product_data['basename'];
+			$button['action'] = 'activate';
+			/* translators: The extension name. */
+			$button['button_text'] = sprintf( __( 'Activate %s', 'easy-digital-downloads' ), $product_data['title'] );
+		}
+
+		return $button;
+	}
+
+	/**
+	 * Gets the upgrade URL for the button.
+	 *
+	 * @todo add UTM parameters
+	 * @since 2.11.x
+	 * @param array $product_data The array of provided data about the extension.
+	 * @param int   $item_id      The item/product ID.
+	 * @param bool  $has_access   Whether the user already has access to the extension (based on pass level).
+	 * @return string
+	 */
+	private function get_upgrade_url( $product_data, $item_id, $has_access = false ) {
+		$url            = 'https://easydigitaldownloads.com';
+		$tab            = ! empty( $_GET['tab'] ) ? sanitize_text_field( $_GET['tab'] ) : '';
+		$slug           = ! empty( $product_data['slug'] ) ? $product_data['slug'] : '';
+		$utm_parameters = array(
+			'p'            => urlencode( $item_id ),
+			'utm_source'   => 'settings',
+			'utm_medium'   => urlencode( $tab ),
+			'utm_campaign' => 'admin',
+			'utm_term'     => urlencode( $slug ),
+		);
+
+		if ( $has_access ) {
+			$url = 'https://easydigitaldownloads.com/your-account/your-downloads/';
+			unset( $utm_parameters['p'] );
+		} elseif ( ! empty( $product_data['upgrade_url'] ) ) {
+			$url = esc_url_raw( $product_data['upgrade_url'] );
+			unset( $utm_parameters['p'] );
+		}
+
+		return add_query_arg(
+			$utm_parameters,
+			$url
+		);
+	}
+
+	/**
+	 * Gets the array of parameters for the link to configure the extension.
+	 *
+	 * @since 2.11.x
+	 * @param array  $product_data  The array of provided data about the extension.
+	 * @return array
+	 */
+	protected function get_link_parameters( $product_data ) {
+
+		if ( empty( $product_data['tab'] ) && empty( $product_data['section'] ) ) {
+			return array(
+				/* translators: the plural Downloads label. */
+				'button_text' => sprintf( __( 'View %s', 'easy-digital-downloads' ), edd_get_label_plural() ),
+				'href'        => add_query_arg(
+					array(
+						'post_type' => 'download',
+					),
+					admin_url( 'edit.php' )
+				),
+			);
+		}
+		return array(
+			/* translators: The extension name. */
+			'button_text' => sprintf( __( 'Configure %s', 'easy-digital-downloads' ), $product_data['title'] ),
+			'href'        => add_query_arg(
+				array(
+					'post_type' => 'download',
+					'page'      => 'edd-settings',
+					'tab'       => urlencode( $product_data['tab'] ),
+					'section'   => urlencode( $product_data['section'] ),
+				),
+				admin_url( 'edit.php' )
+			),
+		);
+	}
```

Comment:
```suggestion
	protected function get_link_parameters( $product_data ) {
		if ( empty( $product_data['tab'] ) && empty( $product_data['section'] ) ) {
			return array(
				/* translators: the plural Downloads label. */
				'button_text' => sprintf( __( 'View %s', 'easy-digital-downloads' ), edd_get_label_plural() ),
				'href'        => add_query_arg(
					array(
						'post_type' => 'download',
					),
					admin_url( 'edit.php' )
				),
			);
		}

		return array(
			/* translators: The extension name. */
			'button_text' => sprintf( __( 'Configure %s', 'easy-digital-downloads' ), $product_data['title'] ),
			'href'        => add_query_arg(
				array(
					'post_type' => 'download',
					'page'      => 'edd-settings',
					'tab'       => urlencode( $product_data['tab'] ),
					'section'   => urlencode( $product_data['section'] ),
				),
				admin_url( 'edit.php' )
			),
		);
	}
```

Silly minor formatting thing:

1. Removed empty line on line 245.
2. Added empty line between 257 and 258.